ssl/quic/quic_impl.c

```diff
@@ -1463,7 +1475,21 @@ long ossl_quic_ctx_ctrl(SSL_CTX *ctx, int cmd, long larg, void *parg)
 
 long ossl_quic_callback_ctrl(SSL *s, int cmd, void (*fp) (void))
 {
-    return ssl3_callback_ctrl(s, cmd, fp);
+    QUIC_CONNECTION *qc = QUIC_CONNECTION_FROM_SSL(s);
+
+    if (!expect_quic_conn(qc))
+        return 0;
+
+    switch (cmd) {
+    case SSL_CTRL_SET_MSG_CALLBACK:
+        qc->msg_callback = (ossl_msg_cb)fp;
```

Comment:
Does this not need to be passed down to the channel etc. also?

Reply:
It seems it does now! Previously this was working because the creation of the channel was deferred until later. I've now rebased and it no longer works because the channel is created much earlier so setting the callback doesn't get propagated down to the other objects. Fix in progress.

Reply:
Fixed.